src/bundle/Controller/LocationController.php

```diff
@@ -354,7 +354,7 @@ public function trashAction(Request $request): Response
             }
         }
 
-        return $this->redirect($this->generateUrl('ezplatform.trash.list'));
+        return $this->redirectToLocation($form->getData()->getLocation());
```

Comment:
why do you need to change behavior?

Reply:
The user tried to move the location to the trash but the operation failed so he should stay on the same page and see the error message. Another issue is that we do not know if the user has permissions to see the trash list.

Reply:
IMHO this fragment should looks as follow: 

```php
if (($form->getData()->getLocation()) !== null)
    return $this->redirectToLocation($location);
}

return $this->redirectToRoute('ezplatform.trash.list');
```

Even if reaching `$this->redirectToLocation($location);` instruction is not possible from UI, the request shouldn't end-up with "500: Internal Server Error".

Reply:
@adamwojs fixed